src/helper/withRouter.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+
+export interface WithRouterProps {
+  location: ReturnType<typeof useLocation>;
+  params: Record<string, string>;
+  navigate: ReturnType<typeof useNavigate>;
+}
+
+export const withRouter = <Props extends WithRouterProps>(
+  Component: React.ComponentType<Props>
+) => {
+  return (props: Omit<Props, keyof WithRouterProps>) => {
+    const location = useLocation();
+    const params = useParams();
```

Comment:
According to [react-router-dom reference](https://reactrouter.com/en/main/hooks/use-params) "useParams" is hook.